tests/data/huggingface/run_tf.py

```diff
@@ -57,10 +57,8 @@
     )
     train_dataset.set_format(type="tensorflow", columns=["input_ids", "attention_mask", "label"])
 
-    train_features = {
-        x: train_dataset[x].to_tensor(default_value=0, shape=[None, tokenizer.model_max_length])
-        for x in ["input_ids", "attention_mask"]
-    }
+    train_features = {x: train_dataset[x] for x in ["input_ids", "attention_mask"]}
```

Comment:
Can you detail why we are removing the to_tensor call, especially since the shape's `tokenizer.model_max_length` parameter is something that we gave explicitly in the past? Is this driven by the change in TF version?

Please make this backwards compatible ie. have an original test case with the previous changes, and add a new test case for this requirement where the to_tensor call is not required so that we test both scenarios.

Reply:
It has been removed in `Datasets` since we changed the internal structure. it used to return RaggedTensor even when the tensors were normal dense tensors. 
And the `tokenizer.mode_max_length` is already represented in 
```python
    train_dataset = train_dataset.map(
        lambda e: tokenizer(e["text"], truncation=True, padding="max_length"), batched=True
    )
``` 
which creates already a shape of `max_length`. 

I added a condition to test to check the `transformers` version and added the old code